File: Protogen/loxim_protocol/cpp/protocol/packages/A_sc_byePackage.cpp
#include <string.h>
#include <stdlib.h>
#include "A_sc_byePackage.h"
#include "ptools/PackageBufferWriter.h"
#include "ptools/PackageBufferReader.h"

using namespace protocol;

A_sc_byePackage::A_sc_byePackage(\
		enum Bye_reasonsEnum a_reason_code,\
		CharArray* a_reason_description
		)
{
	reason_code=a_reason_code;
	reason_description=a_reason_description;
}

A_sc_byePackage::A_sc_byePackage()
{
	reason_description=NULL;
}

A_sc_byePackage::~A_sc_byePackage()
{
	if (reason_description) delete reason_description;
}

void A_sc_byePackage::serializeW(PackageBufferWriter *writer)
{
	Package::serializeW(writer);
	uint32_t tmp_reason_code;
	Bye_reasonsFactory::getValueByEnum(reason_code, tmp_reason_code);
	writer->writeUint32(tmp_reason_code);
	writer->writeString(reason_description);
}

bool A_sc_byePackage::deserializeR(PackageBufferReader *reader)
{
	if (!Package::deserializeR(reader)) return false;
	Bye_reasonsFactory::getEnumByValue(reader->readUint32(),reason_code);
	reason_description=reader->readString();
	return true;
}

bool A_sc_byePackage::equals(Package *_package)
{
	if (!_package) return false;
	if (_package->getPackageType()!=this->getPackageType()) return false;
	if (!Package::equals(_package)) return false;
	A_sc_byePackage* p=(A_sc_byePackage*)_package;
	if (!(getReason_code()==p->getReason_code())) return false;
	if (!(CharArray::equal(getReason_description(),p->getReason_description()))) return false;
	return true;
}

File: Protogen/loxim_protocol/java/src/main/java/pl/edu/mimuw/loxim/protocol/packages/A_sc_byePackage.java
package pl.edu.mimuw.loxim.protocol.packages;

import java.util.EnumSet;
import pl.edu.mimuw.loxim.protogen.lang.java.template.ptools.Package;
import pl.edu.mimuw.loxim.protogen.lang.java.template.exception.ProtocolException;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageInputStreamReader;
import pl.edu.mimuw.loxim.protogen.lang.java.template.streams.PackageOutputStreamWriter;


import pl.edu.mimuw.loxim.protocol.enums.*;


public  class A_sc_byePackage extends Package {
		public final static short ID=(short)3;
	
    	/**
**/
    	private Bye_reasonsEnum reason_code=null;
    	/**
**/
    	private String reason_description=null;
    	
	public A_sc_byePackage()
	{
	}

		public A_sc_byePackage(
		    	Bye_reasonsEnum a_reason_code,
	        	String a_reason_description    	)
	{
        	reason_code=a_reason_code;
        	reason_description=a_reason_description;
    
	}
		
		@Override
	public long getPackageType() {
		return (long)ID;
	}
		
	@Override
	protected void deserializeW(PackageInputStreamReader reader)
			throws ProtocolException {
		super.deserializeW(reader);
		
				try{
	    			long tmp_reason_code=reader.readUint32();
reason_code=Bye_reasonsEnum.createEnumByValue(tmp_reason_code);
					reason_description=reader.readString();
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}
			}
	
	@Override
	protected void serializeW(PackageOutputStreamWriter writer)
			throws ProtocolException {
		super.serializeW(writer);
				try{
	    			writer.writeUint32(reason_code.getValue());		
					writer.writeString(reason_description);		
				}catch(Exception e)
		{
			throw new ProtocolException(e);
		}	
			}
	
	@Override
	public boolean equals(Object obj) {
		if(!super.equals(obj))
			return false;
		A_sc_byePackage p=(A_sc_byePackage)obj;
	    if (!(((reason_code==p.getReason_code())||((reason_code!=null)&&(p.getReason_code()!=null)&&((reason_code).equals(p.getReason_code())))))) 
return false;
		if (!(((reason_description==p.getReason_description())||((reason_description!=null)&&(p.getReason_description()!=null)&&((reason_description).equals(p.getReason_description())))))) 
return false;
				return true;
	}
	
//================= GETTERS AND SETTERS ===================

        	public Bye_reasonsEnum getReason_code(){return reason_code;};
		
    	public void setReason_code(Bye_reasonsEnum a_reason_code){ reason_code=a_reason_code;};
		
        	public String getReason_description(){return reason_description;};
		
    	public void setReason_description(String a_reason_description){ reason_description=a_reason_description;};
		
    
	 
}


